src/main/java/de/refactoringbot/services/main/FileService.java

```diff
@@ -48,63 +55,44 @@
 		}
 		return allJavaFiles;
 	}
-	
+
 	/**
-	 * This method returns all root-folders of java files (like the src folder or
-	 * the src/main/java folder from maven projects)
+	 * This method returns all root folders' absolute paths of the given java files
+	 * (like the src folder or the src/main/java folder of maven projects)
 	 * 
 	 * @param allJavaFiles
 	 * @return javaRoots
 	 * @throws FileNotFoundException
 	 */
 	public List<String> findJavaRoots(List<String> allJavaFiles) throws FileNotFoundException {
```

Comment:
I simplified the method. Please check again to see if I've paid attention to everything correctly.

Reply:
Looks good to me. I added a check to see if the paths actually exist. I had a case where a sample Java file was parsed that included a made-up package name, so the actual directory didn't exist, which caused errors.
The method was originally made by @LHommeDeBat and is used by his classes, so maybe he should take a look as well.